File: algorithms/Replace Elements with Greatest Element on Right Side/Replace Elements with Greatest Element on Right Side.java
class Solution {
    public int[] replaceElements(int[] arr) {
      int greatElement = -1;
        for(int i = arr.length-1; i >= 0; i--) {
            int temp = arr[i];
            arr[i] = greatElement;
            greatElement = Math.max(temp, greatElement);
        }
        return arr;
    }
}

File: scripts/algorithms/R/Replace Elements with Greatest Element on Right Side/Replace Elements with Greatest Element on Right Side.js
 * @param {number[]} arr
 * @return {number[]}
 */
var replaceElements = function(arr) {
    let max = arr[arr.length -1]
    
     for(let j=arr.length - 2; j>=0; --j){
         let curr = arr[j];
         arr[j] = max
         max = Math.max(max,curr)
     }
    
    arr[arr.length -1] = -1;
    return arr;
};